---
title: Age-related gene expression variations <br> in human monocytes and T cells
author:
- Kruthika Iyer
- Cristian Valencia
- Martin Skarzynski
date: "`r Sys.Date()`"
output: slidy_presentation
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, cache = TRUE,
                      include = TRUE, warning = FALSE, message = FALSE)
```

```{r libs}
library(readr)
library(here)
library(plotly)
library(knitr)
```

# Context and Study Design

- The MESA Epigenomics and Transcriptomics Study has been launched to investigate potential gene expression regulatory methylation sites in humans by examining the association between CpG methylation and gene expression in purified human monocytes and T cells from a large study population (community-dwelling participants in the Multi-Ethnic Study of Atherosclerosis (MESA)).
- The MESA Epigenomics and Transcriptomics Study was funded by a National Heart, Lung and Blood Institute grant (R01HL101250) through the NIH Roadmap Epigenomics Program in 2009.
- Data includes transcriptomic and methylomic data from CD4+ samples, collected from 214 individuals.
- Peripheral T cells were isolated from blood (Exam 5) with anti-CD4 coated magnetic beads, and the Illumina HumanHT-12 v4 Expression BeadChip and the Illumina HumanMethylation450 BeadChip were used to provide genome-wide coverage of mRNA expression and DNA methylation, respectively.

# Data
- This data enabled the identification of 
    - CpG loci associated with age (age-DMR) and 
    - age-associated  and expression-associated methylation sites (age-eMS), whose degree of methylation was associated with age and cis-gene expression (+/- 1Mb), after adjusting for other covariates such as race, gender, study site, and sample contaimination with B-cells, monocytes, natural killer cells, and neutrophils.
- To estimate residual sample contamination for monocyte data analysis, separate enrichment scores for
    - neutrophils,
    - B cells,
    - T cells, and
    - natural killer cells.
- The participant race, gender, and study site were provided as 'raceGenderSite variable' which represents a combination of those factors.
- A detailed description of each sample characteristics is included in the 'README.txt'.

# Packages

Processing

- `beadarray` R package
    - QC analyses and bead-type summarization (average bead signal for each type after outlier removal)
- `limma` R package - normal-exponential convolution model analysis to estimate non-negative signal,
    - quantile normalization using all probes (gene and control, detected and not detected) and samples,
    - addition of a recommended (small) offset,
    - log2 transformation and elimination of control probe data from the normalized expression matrix.
- `lumi` package
    - Illumina HumanMethylation450 BeadChip technology employs a two-channel system and uses both Infinium I and II assays; normalization was performed.

Association

- `Stats` package
- `MASS` package

# Goals

# Data Processing

# Results Summary

# Conclusions
